classes/publication/Repository.php

```diff
@@ -661,6 +686,66 @@ public function delete(Publication $publication)
         Hook::call('Publication::delete', [&$publication]);
     }
 
+    /**
+     * Given a Version Stage and a flag of whether the Version isMinor, 
+     * the publication's related data is being updated
+     *
+     * @hook 'Publication::updateVersionData::before' [[ &$newPublication, $publication ]]
+     */
+    public function updateVersionData(Publication $publication, VersionStage $versioningStage, bool $isMinor = true): Publication
+    {
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+        $nextAvailableVersionStage = Repo::submission()->getNextAvailableVersionData($submission, $versioningStage, $isMinor);
+
+        $newPublication = clone $publication;
+        $newPublication->setData('versionIsMinor', $isMinor);
+        $newPublication->setVersionData($nextAvailableVersionStage);
+
+        $newPublication->stampModified();
+        Hook::call(
```

Comment:
For new hooks, `Hook::run` is preferred to `Hook::call`. See: https://github.com/pkp/pkp-lib/discussions/8089